mono/metadata/threads.c

```diff
@@ -2012,11 +2015,13 @@ gint32 ves_icall_System_Threading_WaitHandle_WaitOne_internal(HANDLE handle, gin
 	if (mono_thread_current_check_pending_interrupt ())
 		return map_native_wait_result_to_managed (MONO_W32HANDLE_WAIT_RET_FAILED, 0);
 
+	mono_w32handle_ref(handle);
```

Comment:
@kumpera @alexischr ref inc/dec is atomic, so no need to lock before calling `mono_w32handle_ref` or `mono_w32handle_unref`.